src/fs.register.ts

```diff
@@ -31,9 +32,34 @@ export const fqdn: FinalizeRequestMiddleware<object, MetadataBearer> = (next) =>
   };
 };
 
+/**
+ * AWS SDK middleware logic to try 3 times if receiving an EAI_AGAIN error
+ */
+export function eaiAgainBuilder(timeout: number): BuildMiddleware<object, MetadataBearer> {
+  const eaiAgain: BuildMiddleware<object, MetadataBearer> = (next) => {
+    const maxTries = 3;
+    return async (args) => {
+      for (let i = 0; i < maxTries; i++) {
+        try {
+          return await next(args);
+        } catch (error) {
+          if (error && typeof error === 'object' && 'code' in error && error.code !== 'EAI_AGAIN') {
+            throw error;
+          }
+          logger.error('eai_again:retry:' + i);
```

Comment:
what other information could we log here? eg hostname? would be interesting to see if we are getting the retries on specific hosts only.

I also think this is maybe just a `warn` and `error` when the retries limit is hit.

Reply:
Is there a way to get `error?.hostname` without having to add the very long if statement like in L46 so I don't get this error: `Property 'hostname' does not exist on type '{}'`